qupath-core/src/main/java/qupath/lib/objects/utils/ObjectMerger.java

```diff
@@ -359,6 +359,38 @@ public static ObjectMerger createTouchingMerger() {
                 0);
     }
 
+    /**
+     * Create an object merger that can merge together any objects with sufficiently large intersection over union.
+     * <p>
+     * Objects must also have the same classification and be on the same image plane to be mergeable.
+     * <p>
+     * IoU is calculated using Java Topology Suite intersection, union, and getArea calls.
+     * <p>
+     * This merger assumes that you are using an OutputHandler that doesn't clip to tile boundaries (only to region
+     * requests) and that you are using sufficient padding to ensure that objects are being detected in more than on
+     * tile/region request.
+     * You should probably also remove any objects that touch the regionRequest boundaries, as these will probably be
+     * clipped, and merging them will result in weirdly shaped detections.
+     * @return an object merger that can merge together any objects with sufficiently high IoU and the same classification
+     */
+    public static ObjectMerger createIoUMerger(double iouThreshold) {
+        return new ObjectMerger(
+                ObjectMerger::sameClassTypePlaneTest,
+                createIoUMergeTest(iouThreshold),
+                0.0625);
+    }
+
+    private static BiPredicate<Geometry, Geometry> createIoUMergeTest(double iouThreshold) {
+        return (geom, geomOverlap) -> {
+            var i = geom.intersection(geomOverlap);
```

Comment:
I think `intersection` can be slow - could we benchmark this (ideally through adding tests)?
Since we don't *really* need to get the intersection, I imagine we could use `difference` or `union` instead, and compute the areas from that. But I don't know which of the methods is fastest.

Reply:
I started with `a.intersection(b).getArea() / a.union(b).getArea()` and the profiler was claiming that union was way more expensive than intersection, but I can profile the different options

Reply:
My exceedingly fast and non-systematic test using two QuPath annotations suggests that `intersection` is indeed the fastest - less than half the time of the other two.

```groovy
def geoms = getAnnotationObjects().collect {it.ROI.geometry}
def g1 = geoms[0]
def g2 = geoms[1]

var timeit = new Timeit()
    .microseconds()
    .checkpointAndRun("Checking", () -> {
        g1.intersection(g2)
//        g1.difference(g2)
//        g1.union(g2)
    }, 1000)
    .summarizeCheckpoints()
    .stop();


println(timeit)
```

So.... as you were.